clang/lib/Sema/TreeTransform.h

```diff
@@ -5838,7 +5839,7 @@ QualType TreeTransform<Derived>::TransformDependentAddressSpaceType(
   } else {
     TypeSourceInfo *DI = getSema().Context.getTrivialTypeSourceInfo(
         Result, getDerived().getBaseLocation());
-    TransformType(TLB, DI->getTypeLoc());
+    TLB.TypeWasModifiedSafely(DI->getType());
```

Comment:
I’m candidly not familiar enough with `TypeLocBuilder` to know if this is the correct solution to this, but without this, we now crash if the resulting type *isn’t* dependent, e.g. in
```c++
template <int AS>
using ASPtrTy = void [[clang::address_space(AS)]] *;

ASPtrTy<1> x;
```

Reply:
It does look like it is always safely modified, because in the non-dependent case, we will just add qualifiers to the type, and we don't currently track source locations for qualifiers.

See `BuildAddressSpaceAttr` implementation.

But the `getTrivialTypeSourceInfo` is unnecessary in this case, you can remove it.:

```suggestion
    TLB.TypeWasModifiedSafely(Result);
```

Reply:
> But the `getTrivialTypeSourceInfo` is unnecessary in this case, you can remove it.:

Done

Reply:
Does this pass the existing tests?

Reply:
I ran the tests both after adding the call to `TypeWasModifiedSafely` and after making this change here just now, and they were all fine.